### PR TITLE
mender-cli integration tests

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

**Comment by tranchitella** [comment]
Pipeline (together with https://github.com/mendersoftware/mender-cli/pull/123): https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/274900680

**Review by mchalski — APPROVED** [review]

**Comment by tranchitella** [comment]
waiting for approval from the integration tests superheros...

File: tests/tests/test_filetransfer_cli.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import authentication, devauth, get_container_manager, logger
+from .common_connect import wait_for_connect
+from .common import md5sum
+from .mendertesting import MenderTesting
+
+
+class TestFileTransferCLI(MenderTesting):
+    """Tests the port forward functionality"""
```
Comment (oleorhagen) [inline tests/tests/test_filetransfer_cli.py:30]: Nit:
`Tests the file transfer functionality`

File: tests/tests/test_filetransfer_cli.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout, stderr = p.communicate()
+        exit_code = p.wait()
+        assert exit_code == 0, (stdout, stderr)
+
+        # upload and download files using mender-cli
+        try:
+            # create a 40MB random file
```
Comment (oleorhagen) [inline tests/tests/test_filetransfer_cli.py:75]: Question: 
Can't you just do `os.urandom(40 * 1024 * 1024)` ?

File: tests/tests/test_filetransfer_cli.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                    f.name + ".download",
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            stdout, stderr = p.communicate()
+            exit_code = p.wait()
+            assert exit_code == 0, (stdout, stderr)
+
+            # assert the files is not corrupted
```
Comment (oleorhagen) [inline tests/tests/test_filetransfer_cli.py:121]: Nit:
`files _are_`

**Review by oleorhagen — APPROVED** [review]
Neat!

File: tests/tests/test_filetransfer_cli.py (hunk 2 above)
Comment (tranchitella) [reply tests/tests/test_filetransfer_cli.py:75]: I don't want to consume 40MB of RAM altogether

File: tests/tests/test_filetransfer_cli.py (hunk 1 above)
Comment (tranchitella) [reply tests/tests/test_filetransfer_cli.py]: https://github.com/mendersoftware/integration/pull/1274/commits/e5ec5bb00124b8a5dd5abd5a0ccb23910f9e77fc

**Review by kacf — APPROVED** [review]
Excellent tests!

File: tests/tests/test_filetransfer_cli.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+from .common import md5sum
+from .mendertesting import MenderTesting
+
+
+class TestFileTransferCLI(MenderTesting):
+    """Tests the file transfer functionality"""
+
+    def test_filetransfer_cli(self, standard_setup_one_client):
+        # accept the device
+        devauth.accept_devices(1)
```
Comment (lluiscampos) [inline tests/tests/test_filetransfer_cli.py:34]: I've seen this already in other tests that you @tranchitella write. Is there a good reason for not using `standard_setup_one_client_bootstrapped` instead? The framework will then accept the device and your test only starts after/if is accepted.

File: tests/tests/test_filetransfer_cli.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert 1 == len(devices)
+
+        # wait for the device to connect via websocket
+        auth = authentication.Authentication()
+        wait_for_connect(auth, devices[0])
+
+        # device ID and auth token
+        devid = devices[0]
```
Comment (lluiscampos) [inline tests/tests/test_filetransfer_cli.py:47]: Nitpick: I would do this before lines 43-44 and use it on `wait_for_connect`.

File: tests/tests/test_remote_terminal.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import authentication, devauth, get_container_manager, logger
+from .common_connect import wait_for_connect
+from .common import md5sum
+from .mendertesting import MenderTesting
+
+
+class TestRemoteTerminal(MenderTesting):
+    """Tests the remote terminal functionality"""
```
Comment (lluiscampos) [inline tests/tests/test_remote_terminal.py:30]: :heart:

**Review by lluiscampos — APPROVED** [review]
Excellent :medal_sports: Thank for your the remote terminal tests.

File: tests/tests/test_filetransfer_cli.py (hunk 4 above)
Comment (tranchitella) [reply tests/tests/test_filetransfer_cli.py:34]: uhm, I believe it's a copy-and-paste from other tests. I can try rewriting it using the `standard_setup_one_client_bootstrapped` indeed, let me try it.

Comment (tranchitella) [reply tests/tests/test_filetransfer_cli.py]: Done: https://github.com/mendersoftware/integration/pull/1274/commits/e6d4bff1d6348de853d4fc21b40cfc006af57c62

**Comment by tranchitella** [comment]
@mendeer-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
Failures are hsm related, we can ignore.

**Pull request merged**